src/peakrdl_python/templates/addrmap.py.jinja

```diff
@@ -155,6 +163,34 @@ class {{get_fully_qualified_type_name(node)}}_cls(AddressMap):
         return self.__{{child_node.inst_name}}
             {%- endif %}
         {% endfor %}
+
+    {{ child_getter(node, "get_registers",   "Reg",        systemrdlRegNode) }}
+    {{ child_getter(node, "get_memories",    "Memory",     systemrdlMemNode) }}
+    {{ child_getter(node, "get_addressmaps", "AddressMap", systemrdlAddrmapNode) }}
+    {{ child_getter(node, "get_regfiles",    "RegFile",    systemrdlRegfileNode) }}
+{%- endmacro %}
+
+{%- macro child_getter(node, getter_name, child_type, child_rdltype) %}
+    def {{getter_name}}(self, unroll=False) -> Iterable[Union[{{child_type}},Tuple[{{child_type}},...]]]:
```

Comment:
Hmm. I think there needs to be some means of discovering and getting hold of an array of things. It could be handled by `Node`s having an `is_array_member` property and then a `get_enclosing_array`.

Or we could change the default of `unroll` to `True`, and provide the property too by:
```
    registers = property(fget=get_registers)
```

Reply:
Unlike the system RDL, in the python API the array has no instance, each element of the array is a separate instance within its parent. 

What is the use case you have in mind for accessing the array but not its individual elements?

Reply:
I have started trying to write some examples:
- Dumping all readable registers to a json file (having first put them in a list/dictionary structure)
- Resetting all the writable registers to their default values.

I am starting to appreciate the issue you have been talking about. It is highlighting a few other practical issues. I will add these examples into the documentation as part of this branch so you can take a look.

Reply:
Sorry, I've not been keeping up for the last couple of days (in which you seem to have done an awful lot!).
Yes, dumping to JSON is a good example of where you'd want to treat arrays as arrays.

Reply:
There are two new examples under `example/tranversing_address_map`. I think these address the points you were trying to make. Please let me know what you think, I need to some test cases to check out the generators before we merge this pull request.